src/Components/TypedText.jsx

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+const TypedText = ({string, typeSpeed}) => {
+  const [currentText, setCurrentText] = useState('');
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const cursor = currentIndex < string.length ? "|" : "";
+
+  function moveChar() {
+    setCurrentText(prevText => prevText + string[currentIndex]);
+    setCurrentIndex(prevIndex => prevIndex + 1);
+  }
+
+  useEffect(()=> {
+    if (currentIndex < string.length) {
+      const timeout = setTimeout(moveChar, typeSpeed);
+      return () => clearTimeout(timeout);
+    }
+  }, [currentIndex])
```

Comment:
Hmm, is there a specific reason we use `setTimeout` combined with the dependency-reloading behavior of `useEffect` instead of using a `setInterval`? Also, do we need to `clearTimeout`, since the only time that would be called is if the timeout callback was already executed?

Reply:
used `useEffect` + `setTimeout` because I didn't know `setInterval` existed but that's an easy change. And then yes we need the `clearTimeout` because without it, we prints out "NNrthwestern AutoAquaponics". Not 100% sure why. But also imo it's never a bad idea to clean up async functions just in case.

Reply:
Ah, I see the issue since effects run twice. It's already merged and working, but just for fun here's one that plays nicely with effects running twice (and at arbitrarily slow speeds).
```JS
const TypedText = ({string, typeSpeed}) => {
  const [currentIndex, setCurrentIndex] = useState(0);

  const cursor = currentIndex < string.length ? "|" : "";

  useEffect(()=> {
    let index = 0;
    let interval = setInterval(() => {
      index += 1;
      setCurrentIndex(index);
      if (index == string.length) {
        clearInterval(interval);
        interval = null; // set variable to null so that the cleanup function doesn't clear it twice
      }
    }, typeSpeed);
    return () => clearInterval(interval);
  }, []);

  return <>{string.slice(0, currentIndex)}{cursor}</>;
};
```
One (invisible) advantage this has over the current implementation is that the current implementation depends on React's double-effect behavior running very quickly (otherwise we will already have inserted some characters already before the effect cleans up and runs again), whereas for this one, it will just work properly every time the effect runs.